File: src/frontend/src/types/team.ts
export type TeamMemberRole = 'leader' | 'member';

export interface TeamMember {
  id: string;
  team_id: string;
  agent_id: string;
  agent_name: string;
  role: TeamMemberRole;
  joined_at: string;
  level?: number;
  total_xp?: number;
}

export interface TeamStats {
  total_xp: number;
  total_tasks: number;
  level: number;
  member_count: number;
}

export interface Team {
  id: string;
  workspace_id: string;
  name: string;
  description: string | null;
  icon: string;
  created_by: string;
  created_at: string;
  member_count?: number;
  total_xp?: number;
  level?: number;
  members?: TeamMember[];
}

export interface TeamCreate {
  name: string;
  description?: string;
  icon: string;
}

export type ChallengeGoalType = 'events' | 'tasks' | 'xp';
export type ChallengeStatus = 'active' | 'completed' | 'failed';
export type ChallengeType = 'team' | 'workspace';

export interface Challenge {
  id: string;
  workspace_id: string;
  name: string;
  description: string;
  type: ChallengeType;
  goal_type: ChallengeGoalType;
  goal_value: number;
  current_value: number;
  reward_tokens: number;
  reward_xp: number;
  start_at: string;
  end_at: string;
  status: ChallengeStatus;
}

export interface ChallengeProgress {
  id: string;
  challenge_id: string;
  contributor_id: string;
  contributor_name?: string;
  contribution: number;
  updated_at: string;
}

File: src/backend/app/schemas/team.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List


class TeamCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = None
    icon: str = "users"


class TeamMemberRead(BaseModel):
    id: str
    team_id: str
    agent_id: str
    role: str
    joined_at: datetime

    model_config = {"from_attributes": True}


class TeamRead(BaseModel):
    id: str
    workspace_id: str
    name: str
    description: Optional[str] = None
    icon: str
    created_by: str
    created_at: datetime
    members: List[TeamMemberRead] = []

    model_config = {"from_attributes": True}


class TeamStats(BaseModel):
    team_id: str
    team_name: str
    member_count: int
    total_xp: int
    total_tasks: int
    level: int
    level_name: str


class AddMemberRequest(BaseModel):
    agent_id: str
    role: str = "member"
